OSOE-598: Add entra authentication for live metrics

AydinE opened the pull request:
[OSOE-598](https://lombiq.atlassian.net/browse/OSOE-598)
Fixes #56

[OSOE-598]: https://lombiq.atlassian.net/browse/OSOE-598?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

AydinE committed:
Add entra authentication for live metrics

AydinE committed:
Merge branch 'dev' into issue/OSOE-598

Piedone commented on Readme.md:
```diff
@@ -86,19 +86,30 @@ The module has its own configuration for further options. These need to come fro
   },
   "OrchardCore": {
     "Lombiq_Hosting_Azure_ApplicationInsights": {
+      // Deprecated, do not use in new projects 
```
```suggestion
      // Deprecated, do not use in new projects.
```

Piedone commented on Readme.md:
```diff
@@ -86,19 +86,30 @@ The module has its own configuration for further options. These need to come fro
+
+To set up Entra Authentication follow the steps that most closely match your situation over at [Microsoft Entra authentication for Application Insights](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication?tabs=net)
+
+Once Entra Authentication is set up and the ConnectionString has been properly set, live metrics should be flowing in.
```
```suggestion
Starting 30 September 2025, authentication using API keys is no longer supported. Instead, you'll have to set up Entra Authentication.

To set up Entra Authentication follow the steps that most closely match your situation over at [Microsoft Entra authentication for Application Insights](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication?tabs=net)

Once Entra Authentication is set up and the `ConnectionString` has been properly set, live metrics should be flowing in.
```

Piedone commented on Readme.md:
````diff
@@ -86,19 +86,30 @@ The module has its own configuration for further options. These need to come fro
 ```
 
-See the [`ApplicationInsightsOptions` class](Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs) for all options and details. We recommend configuring at least `QuickPulseTelemetryModuleAuthenticationApiKey`.
+> ⚠ Use of QuickPulseTelemetryModuleAuthenticationApiKey is deprecated and will be officially unsupported starting September 30, 2025. See [Entra Authentication](#entra-authentication) for more information.
````
```suggestion
> ⚠ Use of QuickPulseTelemetryModuleAuthenticationApiKey is deprecated and will be officially unsupported starting 30 September 2025. See [Entra Authentication](#entra-authentication) for more information.
```

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs:
```diff
@@ -28,6 +28,7 @@ public class ApplicationInsightsOptions
     /// documentation for more info: <see
     /// href="https://docs.microsoft.com/en-us/azure/azure-monitor/app/live-stream#secure-the-control-channel"/>.
     /// </summary>
+    [Obsolete("Microsoft Entra authentication is the only supported method from September 30, 2025. API key authentication will be removed")]
```
```suggestion
    [Obsolete("Microsoft Entra authentication is the only supported method from 30 September 2025. API key authentication will be removed.")]
```

Piedone commented on Readme.md:
Please also update the NuGet to the latest v2.22.0 so we can be sure we don't miss any related fixes.

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/Extensions/ApplicationInsightsInitializerExtensions.cs:
```diff
@@ -23,6 +24,13 @@ public static class ApplicationInsightsInitializerExtensions
+
+        services.Configure<TelemetryConfiguration>(config =>
+        {
+            var credential = new ManagedIdentityCredential();
```
We also need local authentication for local development, and the Readme extended with how you'd set that up. I.e., I'm a developer, adding this module to my app (or running it as part of OSOCE), what should I do to make the quick pulse telemetry channel work locally? (I'll also link to this from the Confluence wiki of the project.)

Conversely, what should I do (differently) when deploying it to staging/prod?

Piedone replied:
So, please document this. This isn't enough:

> To set up Entra Authentication follow the steps that most closely match your situation over at [Microsoft Entra authentication for Application Insights](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication?tabs=net)

One needs to dig into the Microsoft documentation to figure out where to start. Add docs pointing people to specifics: one scenario is running this locally, and the other is running it in an Azure service. The second case can be anything but if this matters, point to the most common use cases of Azure App Services and Container Instances.

AydinE replied:
I've updated it to be more straightforward and also explain the setup for local development. Let me know if more changes are needed

AydinE committed:
Update Readme.md

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

AydinE committed:
Update Readme.md

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

AydinE committed:
Update Readme.md

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

AydinE committed:
Update Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

AydinE committed:
Support local development

AydinE committed:
Merge branch 'issue/OSOE-598' of https://github.com/Lombiq/Orchard-Azure-Application-Insights into issue/OSOE-598

AydinE committed:
Update Readme.md

AydinE committed:
Update ApplicationInsightsInitializerExtensions.cs

AydinE committed:
Update nuget package version

AydinE commented:
I've now added a way to use it during local development and added some extra info about it in the readme. I've made it an option that needs to be enabled from the appsettings as it seems that it's in line with how other options for this module are configured.

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/Extensions/ApplicationInsightsInitializerExtensions.cs:
```diff
@@ -43,6 +36,20 @@ public static class ApplicationInsightsInitializerExtensions
+            if (applicationInsightsOptions.EnableLocalDevelopment)
+            {
+                var credential = new DefaultAzureCredential();
+                config.SetAzureTokenCredential(credential);
```
This can be put outside of the `if`.

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs:
```diff
@@ -70,6 +70,14 @@ public class ApplicationInsightsOptions
     public bool EnableClientSideTracking { get; set; } = true;
 
+    /// <summary>
+    /// Gets or sets a value indicating whether local development is enabled where telemetry is sent to
+    /// Azure Application Insights from the local instance.
+    /// Will only work if local authentication is enabled in the properties of the Applications Insights resource
+    /// which should only be the case during development and testing.
```
Keep to https://lombiq.atlassian.net/wiki/spaces/DEV/pages/28213305/General+code+conventions+and+development+practices#:~:text=For%20legibility%20try%20to%20limit%20lines%20of%20code%20to%20120%20characters%20(soft%20limit)%20or%20150%20at%20most%20(hard%20limit%2C%20code%20analyzers%20will%20warn%20about%20it%20too)%2C%20if%20breaking%20at%20120%20would%20hurt%20compactness%20or%20legibility.

Piedone replied:
This isn't addressed.

AydinE replied:
> This isn't addressed.

Oh I forgot to ask you about this, I don't really know what you meant by this because none of these lines exceed this rule the largest line is 114 characters

Piedone replied:
They're shorter though. and the first two lines can be packed tighter, without going over the limit. The Rewrap extension will reformat as such.

AydinE replied:
Understood, should be addressed in the most recent push

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/Extensions/ApplicationInsightsInitializerExtensions.cs:
```diff
@@ -43,6 +36,20 @@ public static class ApplicationInsightsInitializerExtensions
+        {
+            if (applicationInsightsOptions.EnableLocalDevelopment)
+            {
+                var credential = new DefaultAzureCredential();
```
Won't this try all, including `ManagedIdentityCredential`? https://learn.microsoft.com/en-us/dotnet/api/azure.identity.defaultazurecredential?view=azure-dotnet

How did you test this module before this change locally?

AydinE replied:
Changed this to use `DefaultAzureCredential` this was described a bit weird on the ms docs.

AydinE committed:
Update readme and local development

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/Extensions/ApplicationInsightsInitializerExtensions.cs:
```diff
@@ -38,16 +38,11 @@ public static class ApplicationInsightsInitializerExtensions
         services.Configure<TelemetryConfiguration>(config =>
         {
-            if (applicationInsightsOptions.EnableLocalDevelopment)
+            if (!applicationInsightsOptions.EnableLocalDevelopment)
```
We also need authentication during local development. "Local authentication" (somewhat confusingly named) means that using an Entra ID identity is not required (though allowed) and you can use API keys and connection strings without Entra ID. It doesn't mean that the control channel will be authenticated without anything else during local development.

You can now check out the [Local OSOCE AI resource](https://portal.azure.com/#@lombiq.com/resource/subscriptions/f04789ac-5112-4bb1-8725-eb44038c5273/resourcegroups/OpenSourceOrchardCoreExtensions-EUS/providers/microsoft.insights/components/OpenSourceOrchardCoreExtensions-EUS-Local/quickPulse). If you see this, then the control channel is not authenticated:


https://github.com/Lombiq/Orchard-Azure-Application-Insights/assets/1976647/6c90abf6-19fb-4f05-8ea4-e4d087310968

Note that securing this control channel, i.e. the AI account in the Azure Portal telling the app what telemetry to send, is the point of this issue. Telemetry, including live metrics, can be sent from the app to AI without this too (as seen happening here as well).

AydinE replied:
I have looked into this more today. 

If you disable the confusingly named `LOCAL AUTHENTICATION` the only way to authenticate the local client to the AI resource is setting up a Service Principal (One of 2 options described [here](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication?tabs=net#configure-and-enable-microsoft-entra-id-based-authentication))

The documentation in the readme and naming I have slightly changes to hopefully make things more clear.

The way I understand it is that it comes down to basically 4 scenario's now:

1. If you have `Local Authentication` enabled, all you need to do is supply the connectionstring (Not Authenticated)
2. If you have `Local Authentication` disabled, you need to set up a service principal and supply the details in order to securely push metrics from a locally hosted service.
3. For a service hosted on Azure (AppService, VM or other) you need to set up a Managed Identity, supply it with the Metrics Publisher Role and with that the `DefaultAzureCredential` will work to publish metrics securely. 
4. For now also supporting QuickPulseTelemetryModuleAuthenticationApiKey which is the deprecated method.

These are the scenario's I have tested today, please let me know if I have missed anything.

AydinE replied:
In reply to: https://github.com/Lombiq/Orchard-Azure-Application-Insights/pull/85#discussion_r1628492342
Receiving Live Metrics with just connection string does still work as described in scenario 1.

> Both b) and c) satisfy Local Authentication being disabled. Right? Or is b) still Local Authentication?

Only service principal can work locally because you can't have a managed identity locally. 

After testing these scenario's again today I've gone back in and changes some things around. While Service Principal will be the only way to authenticate during "local" development (or non-azure). It is also possible to use it inside of Azure if for some reason this is preferred over the recommended Managed Identity.

I've now renamed some things, updated the readme and mostly stopped referencing "Local Development" because it causes some confusion.

With regards to:

> Do I understand correctly that under https://github.com/Lombiq/Orchard-Azure-Application-Insights/pull/85#discussion_r1627908324 you mention that with Local Authentication disabled you need to use a Service Principal or Managed Identity even if you don't want to secure the control channel?

This was actually the way that I originally understood it, but it seems that this is not the case. This also caused additional confusion. I think with the latest version this should be mostly cleared up though

Piedone replied:
> Receiving Live Metrics with just connection string does still work as described in scenario 1.

Just the connection string doesn't work, I tried before my last comment, no Live Metrics. I guess because if you don't set up SP credentials, then `DefaultAzureCredential` will be used, which won't work locally and breaks it.

> Only service principal can work locally because you can't have a managed identity locally.

Yes, but that's not what I'm asking. My question is, if Local Authentication is disabled, will an SP work, or only Managed Identity? So, together with your last paragraph please clear up in the docs and config names the following scenarios: What should I configure in the app, and what should I configure in the Azure Portal (can I disable Local Authentication)?

1.  I want to only ingest data in AI and don't care about the control channel. This is most local development scenarios but can very well be for deployed apps too. I think this scenario is the same for any hosting environment (you just have to provide the connection string) but please check.
2.  I want to both ingest data and secure the control channel, in an environment where Managed Identity is NOT available (locally or Azure Services not supporting it).
3.  I want to both ingest data and secure the control channel, in an environment where Managed Identity is available.

AydinE replied:
I made some changes today to ensure that using just connectionstring still works without any changes.
I did have to introduce another option into the appsettings though because I couldn't see any other way to support just using connectionstring without it.

I have also made more edits to the readme to hopefully make it more clear what needs to be set up in which scenario.

To recap:
1) This will work with just connectionstring as long as `Local Authentication` is enabled. I've also tried to make this more clear in the readme
2) This requires a SP, again I've tried to make this more clear in the readme now.
3) This now requires `"UseEntraAuthentication": true` in the appsettings, the readme is updated to reflect this.

If the readme is still not clear enough maybe we can just add these 3 scenario's as a sort of FAQ style section at the bottom of the readme something like "Frequent scenario's" and the exact steps to take?

Piedone replied:
Great, then! I don't think a FAQ, specifically, is needed, but please emphasize which option is necessary/available in which scenario (understanding that people will come with a scenario in mind, a use case, not an understanding of what Local Authentication/Service Principals/Managed Identity is).

Also add validation to throw an exception with an invalid combination of settings (like you can't have SP and MI at the same time).

AydinE replied:
> Also add validation to throw an exception with an invalid combination of settings (like you can't have SP and MI at the same time).

I am not entirely sure how I would go about this. If you enabled `UseEntraAuthentication` but have not enabled `UseServicePrincipalAuthentication` it will default to try `DefaultAzureCredential` which will fail silently if it can't resolve any and you just will not see any metrics.

However if you have `UseEntraAuthentication`  and `UseServicePrincipalAuthentication` it will try the SP credentials this will already throw an error if it's not a valid configuration.

The only extra check I can do is to see if you have 'UseServicePrincipalAuthentication' enabled without having `UseEntraAuthentication`  enabled - which would be an invalid configuration.

Piedone replied:
Perhaps UseEntraAuthentication should then be more like UseManagedIdentityAuthentication? And it's either-or, and both are Entra. Though at this point rather having an enum for the two and a single property would be better.

Piedone commented on Readme.md:
```diff
@@ -86,19 +86,50 @@ The module has its own configuration for further options. These need to come fro
 
+### Entra Authentication
+
+Starting 30 September 2025, authentication using API keys is no longer supported. Instead, you'll have to set up Entra Authentication.
```
```suggestion
Starting 30 September 2025, authentication using API keys is no longer supported for [securing the Live Metrics control channel](https://learn.microsoft.com/en-us/azure/azure-monitor/app/live-stream#secure-the-control-channel). Instead, you'll have to set up Entra Authentication for that (you may omit this if not needed; configuring the connection string is necessary in any case, and enough for simply collecting telemetry).
```

Piedone commented on Readme.md:
```diff
@@ -86,19 +86,50 @@ The module has its own configuration for further options. These need to come fro
 
 If you use the security defaults from [Lombiq Helpful Libraries - Orchard Core Libraries - Security](https://github.com/Lombiq/Helpful-Libraries/blob/dev/Lombiq.HelpfulLibraries.OrchardCore/Docs/Security.md), then the security headers necessary to use Application Insight's client-side tracking will automatically be added.
 
+### Entra Authentication
```
Also adjust the link in "Use of QuickPulseTelemetryModuleAuthenticationApiKey is deprecated..." above.

```suggestion
### Entra Authentication for the Live Metrics control channel
```

Piedone commented on Readme.md:
```diff
@@ -86,19 +86,50 @@ The module has its own configuration for further options. These need to come fro
+
+#### Local development
+
+If you want to stream local metrics to Application Insights, you should set the `EnableLocalDevelopment` option to `true` in the `Lombiq_Hosting_Azure_ApplicationInsights` section of your configuration and make sure that `Local Authentication` is enabled in the Application Insights resource on Azure.
```
If there's a better docs

```suggestion
If you want to stream local metrics to Application Insights, you should set the `EnableLocalDevelopment` option to `true` in the `Lombiq_Hosting_Azure_ApplicationInsights` section of your configuration and make sure that `Local Authentication` is enabled in the Application Insights resource on Azure (for that, do the **opposite** of what's documented [here](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication?tabs=net#disable-local-authentication)).
```

Piedone commented on Readme.md:
```diff
@@ -86,19 +86,50 @@ The module has its own configuration for further options. These need to come fro
+
+#### Setting up Entra Authentication for Application Insights
+
+To set up Entra Authentication for an application hosted on Azure you will have to set up a Managed Identity for the application and give it the `Monitoring Metrics Publisher` role to be able to publish metrics to AI. See how to set up a managed identity [here](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/managed-identities-status). See more on Assigning Azu ...
```
```suggestion
To set up Entra Authentication for an application hosted on Azure you will have to set up a Managed Identity for the application and give it the `Monitoring Metrics Publisher` role to be able to publish metrics to AI. See how to set up a managed identity [here](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/managed-identities-status). See more on Assigning Azure roles [here](https://learn.microsoft.com/en-us/azure/role-based-access-control/role-assignments-portal).
```

AydinE committed:
Update Readme.md

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

AydinE committed:
Update Readme.md

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

AydinE committed:
Update Readme.md

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

AydinE committed:
Update Readme.md

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

AydinE committed:
Use service principal for local development

AydinE committed:
Update ApplicationInsightsOptions.cs

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs:
```diff
@@ -71,12 +71,26 @@ public class ApplicationInsightsOptions
+    /// tenantId, clientId, and clientSecret of the service principal.
     /// </summary>
-    public bool EnableLocalDevelopment { get; set; }
+    public bool EnableSecureLocalDevelopment { get; set; }
```
To follow what AI does, I'd rather use:

```suggestion
    public bool EnableSecureControlChannelLocalAuthentication { get; set; }
```

And explain in the comment, with a link, what it is.

But! I'm confused about whether using the credentials of a Service Principal is considered Local Authentication or not (a Managed Identity definitely isn't). Here I assumed it is, but I'm not sure, please check.

Let's first finish the convo about this below: https://github.com/Lombiq/Orchard-Azure-Application-Insights/pull/85#discussion_r1628492342

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs:
```diff
@@ -71,12 +71,26 @@ public class ApplicationInsightsOptions
+    public bool EnableSecureLocalDevelopment { get; set; }
+
+    /// <summary>
+    /// Gets or sets the tenant ID of the Azure AD application that's used for local development.
```
"Azure AD" is the old name.

```suggestion
    /// Gets or sets the tenant ID of the Microsoft Entra application used to secure the control channel.
```

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs:
```diff
@@ -71,12 +71,26 @@ public class ApplicationInsightsOptions
+    public string TenantId { get; set; }
+
+    /// <summary>
+    /// Gets or sets the client ID of the Azure AD application that's used for local development.
```
```suggestion
    /// Gets or sets the client ID of the Microsoft Entra application used to secure the control channel.
```

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/Extensions/ApplicationInsightsInitializerExtensions.cs:
```diff
@@ -38,7 +38,15 @@ public static class ApplicationInsightsInitializerExtensions
         services.Configure<TelemetryConfiguration>(config =>
         {
-            if (!applicationInsightsOptions.EnableLocalDevelopment)
+            if (applicationInsightsOptions.EnableSecureLocalDevelopment)
```
We need to support the scenario of Live Metrics receiving data without a secure control channel (since you only need that for control, not for ingestion). Just configuring the connection string should still work.

Then, if you want to secure the control channel, you can optionally do that with a) API key (deprecated), b) secrets with a Service Principal or c) with a managed identity when deployed to Azure. Note that while we need b) for local _development_ too, it's actually not just for that, since you can use Service Principals in Azure too (you may actually need that for services not supporting Managed Identity).

Both b) and c) satisfy Local Authentication being disabled. Right? Or is b) still Local Authentication?

Do I understand correctly that under https://github.com/Lombiq/Orchard-Azure-Application-Insights/pull/85#discussion_r1627908324 you mention that with Local Authentication disabled you need to use a Service Principal or Managed Identity even if you don't want to secure the control channel?

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs:
```diff
@@ -71,12 +71,26 @@ public class ApplicationInsightsOptions
+    public string ClientId { get; set; }
+
+    /// <summary>
+    /// Gets or sets the client secret of the Azure AD application that's used for local development.
```
```suggestion
    /// Gets or sets the client secret of the Microsoft Entra application used to secure the control channel.
```

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs:
```diff
@@ -71,12 +71,26 @@ public class ApplicationInsightsOptions
+    /// <summary>
+    /// Gets or sets the tenant ID of the Azure AD application that's used for local development.
+    /// </summary>
+    public string TenantId { get; set; }
+
+    /// <summary>
+    /// Gets or sets the client ID of the Azure AD application that's used for local development.
+    /// </summary>
+    public string ClientId { get; set; }
+
+    /// <summary>
+    /// Gets or sets the client secret of the Azure AD application that's used for local development.
+    /// </summary>
+    public string ClientSecret { get; set; }
```
These three configs are just for local auth of the control channel. So, not to pollute this type, it'd be better to put them into a separate class, like `SecureControlChannelLocalAuthenticationCredentials` and have a property with that, named the same.

Piedone reviewed (changes requested):
I'm still confused about how the whole thing is supposed to work with Azure, sorry about that. In either case, something is missing.

AydinE committed:
Update Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

AydinE committed:
Update Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

AydinE committed:
Update Lombiq.Hosting.Azure.ApplicationInsights/ApplicationInsightsOptions.cs

Co-authored-by: Zoltán Lehóczky <zoltan.lehoczky@lombiq.com>

AydinE committed:
Try to clear up some confusion

AydinE committed:
Add option to use entra

AydinE committed:
Update Readme.md

Piedone commented on Lombiq.Hosting.Azure.ApplicationInsights/Extensions/ApplicationInsightsInitializerExtensions.cs:
```diff
@@ -35,6 +36,24 @@ public static class ApplicationInsightsInitializerExtensions
             .GetSection("OrchardCore:Lombiq_Hosting_Azure_ApplicationInsights");
         applicationInsightsConfigSection.Bind(applicationInsightsOptions);
 
+        if (applicationInsightsOptions.UseEntraAuthentication)
```
Use brackets for such larger `if` blocks, even if it's a single expression, for clarity.

AydinE committed:
Add validation check

AydinE committed:
Change options to use Enum

Piedone committed:
Grammar

Piedone committed:
JSON syntax fix

Piedone commented on Readme.md:
```diff
@@ -108,33 +108,49 @@ Starting 30 September 2025, authentication using API keys is no longer supported
 
+> ⚠ This section is required if you have disabled `Local Authentication` on your AI resource, See [the docs](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication?WT.mc_id=Portal-AppInsightsExtension&tabs=net#disable-local-authentication).
+
+If you want to use Entra Authentication for Application Insights, or if you have disabled `Local Authentication` on your AI resource, you will have to set up the `EntraAuthenticationType` option to the authentication type you want to use (`ManagedIdentity` or `ServicePrincipal`) in the `Lombiq_Hosting_Azure_ApplicationInsights` section of your configuration like below.
```
Even after disabling Local Auth with `"EntraAuthenticationType": "None"` telemetry is being ingested for me just with a connstring, including Live Metrics working (not telemetry filtering, that needs the secure control channel). So, it perhaps only affects the API key? Please check.

AydinE replied:
I have also noticed this behavior before which just caused more confusion, however it doesn't seem to be working consistently and is also not in line with Microsofts own docs.

Perhaps this is due to a grace period or something but as per the docs, what is currently described in the readme is the correct way going forward, so I would suggest keeping it like this. Let me know what you think

AydinE replied:
> While testing the setup, I clarified the docs directly instead of doing a dozen code suggestions, but please check.

I am not exactly sure what you mean by this

Piedone replied:
OK then!

Just that I pushed to the repo.

Piedone replied:
Ingesting telemetry with local auth disabled still works for me after 3 days... But whatever, it really shouldn't, nothing to do.

Piedone committed:
Clarifying docs

Piedone reviewed (changes requested):
While testing the setup, I clarified the docs directly instead of doing a dozen code suggestions, but please check.

AydinE committed:
Merge branch 'dev' into issue/OSOE-598

Piedone committed:
Not sending QuickPulseTelemetryModuleAuthenticationApiKey if Entra authentication is used

Piedone committed:
Docs
